pkg/gateway/operations/postobject.go

```diff
@@ -39,6 +39,17 @@ func (controller *PostObject) RequiredPermissions(_ *http.Request, repoID, _, pa
 
 func (controller *PostObject) HandleCreateMultipartUpload(w http.ResponseWriter, req *http.Request, o *PathOperation) {
 	o.Incr("create_mpu")
+	branchExists, err := o.Catalog.BranchExists(req.Context(), o.Repository.Name, o.Reference)
+	if err != nil {
+		o.Log(req).WithError(err).Error("could not check if branch exists")
```

Comment:
Not sure we should fail here. Why not just log and keep going? Either the user will succeed or they will fail - so bailing out here only increases the chance of failure when the "impossible" happens.

Reply:
Letting the function continue running causes the server to begin creating and sending parts to be uploaded. This will fail in any case since the branch does not exist. We might as well fail at the first point to avoid unnecessary work.

Reply:
Sorry, I was unclear.  `BranchExists` will _fail_ and return an error when something went wrong.  It doesn't say the branch does or does not exist, only that it does not know.  If the failure was sporadic, we may still succeed.

It's not vital, just failing an informational test should IMO log-and-continue ("best effort").

Reply:
If this is a sporadic error, I don't feel it's that bad if we exit with en error, at max the user will try again. I can fix it if you'd like, it's your call.